cmd/crank/beta/validate/unknown_fields.go

```diff
@@ -0,0 +1,25 @@
+package validate
+
+import (
+	"fmt"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema/pruning"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"strings"
+)
+
+// validateUnknownFields Validates the resource's unknown fields against the given schema and returns a list of errors.
+func validateUnknownFields(mr map[string]interface{}, sch *schema.Structural) field.ErrorList {
+	opts := schema.UnknownFieldPathOptions{
+		TrackUnknownFieldPaths: true, // to get the list of pruned unknown fields
+	}
+	errs := field.ErrorList{}
+
+	unkFlds := pruning.PruneWithOptions(mr, sch, true, opts)
+	for _, f := range unkFlds {
+		strPath := strings.Split(f, ".")
+		child := strPath[len(strPath)-1]
+		errs = append(errs, field.InternalError(field.NewPath(f), fmt.Errorf("unknown field \"%s\"", child)))
```

Comment:
I think we can use a more accurate error type here.  From the [`k8s.io/apimachinery/pkg/util/validation/field` package docs](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation/field#InternalError):

> InternalError returns a *Error indicating "internal error". This is used to signal that an error was found that was not directly related to user input.

I'd argue that an error like this is actually due to user input and they can fix it with better input.  Maybe [`Invalid`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation/field#Invalid) or [`NotFound`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation/field#NotFound) would be a better fit? What do you think, do any of those other error types look more accurate to you? 🤔

Reply:
Actually at first I started with [`Invalid`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation/field#Invalid) but [`fields`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation/field) package's [`ErrorTypes`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation/field#ErrorType) are for value of the field rather than the field name itself. The main reason I selected `Internal Error` was this but looking at it again, I feel like we can change it to [`Invalid`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation/field#Invalid) with an `unknown field ....` extension. Deciding between `Not Found` and `Invalid` I am %51 on the `Invalid` side since `Not Found` may confuse the users e.g. when the typo at the field is a bit hard to see.

Here is the error string when we use `field.Invalid`:
```bash
[x] schema validation error example.crossplane.io/v1beta1, Kind=UserAccess, example : spec.accesKey: Invalid value: "accesKey": unknown field "accesKey"
```
Here is the error string when I use `field.NotFound`:
```bash
[x] schema validation error example.crossplane.io/v1beta1, Kind=UserAccess, example : spec.accesKey: Not found: "accesKey"
```
If one of these error strings is nice I can change the current one to that.

Reply:
I have a preference for the experience you've shown here that uses `field.Invalid`. It may be slightly misusing it from its original intent, but I think it is the closest error type of the package for this scenario. Let's go with that one!